### DISCUSSION TITLE
Verbindungsabbrüche zur DTU

### DISCUSSION THREAD

**Discussion by hoschiking** [hoylabs/OpenDTU-OnBattery#1135] [Q&A] [answered]
Hallo zusammen,
ich habe seit einigen Monaten ein Problem, welches ich nicht eingrenzen kann und komme daher nicht weiter.
Ausgangslage:
Am Anfang hatte ich nur einen HM300, der zur Einspeisung aus der Batterie dient, da ist das Problem nie aufgetreten.
Vor einigen Monaten kam noch ein Balkonraftwerk dazu, eingespeist wird mit einem HM600.
Den habe ich ebenfalls in der OpenDTU-OnBattery eingebunden, damit ich die Daten per MQTT im ioBroker auswerten kann.
Seit dem kommt es zu sporadischen Verbindungsabbrüchen, im Schnitt alle 2-4 Tage.
Ich habe schon alles durchgetestet, modifiziert, nichts hilft.
Der Schwellwert für Erreichbarkeit habe ich schon auf 200 gestellt.
Ich schliesse fast schon aus, dass ein der Wechselrichter solange nicht erreichbar ist (Die Anzeige im Webinterface wechselt auf rot, ab dann ist keiner der beiden Wechselrichter erreichbar und es bleibt so, bis ich OpenDTU per Webinterface neu starte, dann funktioniert alles als wenn nichts gewesen wäre.

Der Fehler ist lästig, da in der Zeit keine Einspeisung mehr erfolgt und ich muss 3 mal täglich es kontrollieren, ob alles läuft.
Den Log, wenn der Fehler auftritt finde ich nicht mehr.....
Nachfolgend ein Auszug aus dem Log, wenn der Fehler nicht auftritt.
[OpenDTU_log_26.07.2024.txt](https://github.com/user-attachments/files/16387568/OpenDTU_log_26.07.2024.txt)

Einer eine Idee?
Mein Dank im Voraus

**Comment by spcqike** [comment]
> Der Fehler ist lästig, da in der Zeit keine Einspeisung mehr erfolgt

wie kommst du drauf? Wird der WR, bevor der Fehler passiert, deaktiviert? 

Ich habe auch hier und da Verbindungsprobleme. Hab alles mögliche probiert. Von Netzteil der DTU über das USB Kabel, über diverse Anpassungen in der Firmware (sendeleistung hoch und runter, Frequenz gewechselt) und die Position der DTU bis hin zu einer neuen DTU (fusion board, anstelle des selbst gelöteten teils)

Bei mir hat es sich irgendwann gebessert. Dennoch hab ich sporadisch aller paar Tage Probleme mit der Verbindung. Bei mir reicht es im Reiter „DTU“ die Einstellungen zu speichern (ohne was daran zu ändern). Dann dauert es keine 5s und das Interface wird wieder blau und alles läuft wie gewohnt für ein paar Tage. 
Daher denke ich, dass es eigentlich ein Software Problem ist. Wundere mich aber, dass so wenig andere User davon berichten. 

Und nein: bei mir läuft der WR unverändert weiter. Ich bekomme zwar keine aktuellen Daten, verständlich, und kann ihn auch nicht per DPL regeln. Er läuft einfach auf seinem zuletzt gesetzten Limit und speist ein.

**Reply by hoschiking** [reply]
weil der Fehler meistens tagsüber auftritt, und genau da ist der Wechselrichter aus, da keine Einspeisung notwendig.
Selbst wenn eine Einspeisung mit dem zuletzt eingestellten Wert erfolgen würde, ist es keinen Deut besser.
Dann speist der der Wechselrichter solange ein, bis der Speicher leer ist. Dann schon lieber gar nicht einspeisen.
In diesem Zustand ist das Fenster für beide Wechselrichter rot, es sind also beide nicht mehr erreichbar.
Da kann ich auch umkonfigurieren was ich möchte, es hilft nur ein Reboot. Das kann auch nicht die Lösung sein.

Ich könnte einen externen Watchdog implementieren und im Fehlerfall ein Reboot erzwingen.
Per Hardware wären die Änderungen sehr aufwändig und nicht wirklich schön, da auf der Platine nicht vorgesehen.
Lässt sich OpenDTU irgendwie per Software rebooten? z.B. mit einem HTTP Request?
Damit wäre der Fehler immer noch nicht behoben, ich hätte aber zumindest vorerst einen Workaround, damit die Anlage läuft

**Reply by spcqike** [reply]
> Software rebooten? z.B. mit einem HTTP Request?

klar. Das webUI macht beim Klick auf „Neustart“ nichts anderes.

**Reply by hoschiking** [reply]
Kannst du mir ein Beispiel nennen wie der Aufruf aussehen soll?
Ich kriege es nicht hin

**Reply by spcqike** [reply, marked as answer]
Ne, kann ich leider nicht direkt. Hab es selbst nirgends implementiert.

guck dir dazu mal die Dokumentation im Upstream an
https://www.opendtu.solar/firmware/web_api/
`<html><body>

Post | yes | /api/maintenance/reboot
-- | -- | --



</body></html>`

**Reply by Michel83** [reply]
Hi, eventuell mal nach der Antenne an der DTU geschaut. Ich hatte ein ähnliches Problem beim umstieg auf die HMS Serie hier war die Antenne die bei dem CMT Modul dabei war einfach zu kurz. 

Verlängert und das Empfangsproblem war erledigt.

https://github.com/tbnobody/OpenDTU/discussions/1879

Grüße

**Reply by hoschiking** [reply]
Alles überprüft, außerdem habe ich die HM-Serie, also ist das Funkmodul ein anderes. Das Thema mit der falschen Antenne ist da nicht vorhanden. In meinen Augen ist es ein Bug, da ich alles andere ausschliessen kann.
Entferne ich aus der config den zweiten Wechselrichter, tritt der Fehler nicht mehr auf.

**Reply by Martin0475** [reply]
Ich habe bemerkt,  wenn der WR (HMS)  nicht mehr erreichbar ist,  dann hat die DTU kurz zuvor neu gestartet.
Siehe Info> System.

Bei mir hilft auch in der Weboberfläche auf die DTU Einstellungen zu gehen und einfach auf speichern klicken ohne zuvor etwas verändert zu haben. 


Möglicherweise wäre es ein Workaround wenn dieses speichern automatisch ausgelöst wird sobald im Log über längere Zeit (All Missing / Radio ok?) erscheint.

Achja: wenn ich die Firmware mit vscode flashe muss ich auch jedesmal erst einmal in den DTU Einstellungen auf speichern klicken. 

Was wird da ausgelöst? Wird der Cmt2300a neu initialisiert? oder ähnliches?

**Reply by schlimmchen** [reply]
> Möglicherweise wäre es ein Workaround wenn dieses speichern automatisch ausgelöst wird sobald im Log über längere Zeit (All Missing / Radio ok?) erscheint.

Ein Auto-Workaround? Eher nicht. Die Ursache muss gefunden und abgestellt werden. Aber diese Meldungen häufen sich und daher muss was dran sein. Ich hab gestern noch den Code gesehen und da ist mir leider nichts aufgefallen...

**Reply by spcqike** [reply]
> Ich hab gestern noch den Code gesehen und da ist mir leider nichts aufgefallen...

Kann es vielleicht an der lib oder am wechselrichter selbst liegen? 

Dass bei irgendwelchen schweinerein in der RF Umgebung etwas passiert bzw. ausgelöst wird, was wir gar nicht beeinflussen können? 

Ich hab die Probleme ja ebenfalls, darüber haben wir schon geschrieben. Nur, in letzter Zeit sind sie wie durch Zauberhand verschwunden. Ich habe weder an der Hardware was geändert noch an der Software (ok jetzt hab ich schon die neuste Version installiert, aber die Probleme haben sich bereits vorher reduziert bzw. sind ganz verschwunden)

**Reply by hoschiking** [reply]
Bei mir ist es reproduzierbar. Solange nur ein WR angebunden ist, keine Probleme, ab 2 fast täglich und nur durch einen reboot der dtu lösbar. Daher würde ich alles rund um EMV ausschließen. Der WR ist auch nach einem reboot bin opendtu sofort erreichbar, somit scheint der auch nicht die Ursache zu sein.

**Reply by schlimmchen** [reply]
Zwei vom gleichen Typ, oder einmal NRF24 und einer CMT2300? Wenn nur einer, dann von welchem Typ?

**Reply by hoschiking** [reply]
Beide NRF24, 1x hm600, 1x hm300

**Comment by Gururom** [comment]
Ich habe  ja, wie auch einige andere auch ab und an solch Problem, bei mir denke ich so alle 1-2 Monate . Irgendwann fängt sich das wieder, ein Reboot der DTU bringt nichts, auch kein Neustart des WR, jedenfalls bei mir.
Habe aber auch nur einen WR.
Ich bin ja dabei den WR zu verdächtigen, das er mit funken aufhört, oder die Frequence verschiebt oder oder oder. die DTU schließe ich eigentlich aus,
In deinem Fall versuchs doch mal mit einem täglichen Neustart der DTU, dass kann man doch glaub ich irgendwo einstellen.
Ansonsten einen ShellyPlug ran und einen täglichen Hardreset machen lassen

**Reply by hoschiking** [reply]
Da das Verhalten bei dir ein anderes ist, gehe ich davon aus, dass die Ursache eine andere ist. Täglicher Reset wäre zwar nicht für Lösung des Problems, aber vielleicht hilft es. Ich wüsste aber nicht, wo man den Reset einstellen kann. Ich glaube, das ist nicht vorgesehen. Ich muss mir was anderes einfallen lassen. 
Der Grund warum ich es hier poste, ist eigentlich der Ursache aus dem Grund zu gehen. Das scheint ein weit verbreitetes Problem zu sein, ohne dass es beseitigt wird.
Wenn's nicht stabil läuft, dann ist das echt Käse

**Reply by spcqike** [reply]
> Das scheint ein weit verbreitetes Problem zu sein, ohne dass es beseitigt wird.
Wenn's nicht stabil läuft, dann ist das echt Käse

Du bist herzlich eingeladen, der Ursache auf den Grund zu gehen, die Stelle im Code zu finden, das potentielle Problem zu beheben und einen PR zu eröffnen. 

Dafür müsste es wohl aber erst einmal reproduzierbar bei unterschiedlichen Installationen auffindbar sein. 

Alle hier beteiligten Personen tun dies in ihrer Freizeit und aus Spaß an der Sache. 

Auch ich habe gelegentlich Verbindungsprobleme. Mal mehr, mal weniger. Da sie bei mir aber sehr unregelmäßig und absolut nicht reproduzierbar sind, denke ich nicht, dass das Grundproblem in der openDTU Firmware liegt. Vielleicht ist es ein merkwürdiges Signal von außerhalb was irgendwas triggert? Ein EMV Problem? Keine Ahnung. 

Die, ich sage mal, „Hauptentwickler“ scheinen das Problem nicht zu haben. Zumindest hab ich diesbezüglich noch nichts gehört. Das macht ein Auffinden des Problems und das Beheben ungemein schwieriger.

**Reply by hoschiking** [reply]
Diesen Post hättest du dir echt sparen können. Warum so negativ?
Was glaubst du warum ich es hier poste und in diesem Projekt aktiv bin?
Natürlich möchte ich die Ursache finden. Ich kann nicht programmieren, Software ist nicht mein Ding.
Ich kann aber testen, Testumgebung zur Verfügung stellen,  also auch sinnvolles beitragen.
All das ergibt sich aus meinen bisherigen Posts, die hast du zum großen Teil auch gelesen.

"Alle hier beteiligten Personen tun dies in ihrer Freizeit und aus Spaß an der Sache" -> auch das trifft auf mich zu.
Also höre bitte auf mich in eine Ecke zu schieben, in die ich nicht hingehöre.

"Auch ich habe gelegentlich Verbindungsprobleme. Mal mehr, mal weniger. Da sie bei mir aber sehr unregelmäßig und absolut nicht reproduzierbar sind, denke ich nicht, dass das Grundproblem in der openDTU Firmware liegt. Vielleicht ist es ein merkwürdiges Signal von außerhalb was irgendwas triggert? Ein EMV Problem? Keine Ahnung"
-> An welcher Stelle hab ich behauptet, dass es an der Firmware liegt? Was du dir im Kopf selbst herbeidichtest kann ich nichts für.

Ich habe ein Problem und versuche es mit Hilfe anderer User einzugrenzen, da ich nicht weiterkomme.
Auch du bist herzlich eingeladen bei der Fehlersuche mitzuwirken, wenn was sinnolles beitragen kannst, gerne.
Dafür wäre ich sehr dankbar

**Comment by Gururom** [comment]
Du hast wohl behauptet das Es an der Software liegt.
"n meinen Augen ist es ein Bug, da ich alles andere ausschliessen kann."
Oder versteh ich das falsch?

**Comment by hoschiking** [comment]
Zeige mir bitte den Post, wo ich geschrieben haben soll, dass es an der Software liegt?

Ein Bug kann auch ein Fehler in einem mechanischen oder elektrischen System sein, nicht nur in Software, wie es die meisten glauben. 
Als Thomas Edison den Begriff erstmals verwendet hat, da gab es noch keine Software

**Reply by Gururom** [reply]
Ok, dann habe ich mich geirrt.!
Vielleicht ist es ja ein wirklicher Bug der die Funkverbindung stört.

**Comment by hoschiking** [comment]
btw:
habe vor einer Woche den zweiten Wechselrichter wieder aus der Liste entfernt, seit dem läuft es wieder stabil.
Dann spontan als Zwischenlösung AhoyDTU auf einem ESP8266 für den zweiten Wechselrichter eingerichtet,
auch dies läuft völlig störungsfrei. Sonst bewußt keine weiteren Änderungen vorgenommen.
Ich bin überzeugt, dass es ein Bug ist, alles was ich mit meinen Mitteln testen konnte brachte keine Besserung
